exp/api/v1beta2/awsmachinepool_webhook_test.go

```diff
@@ -174,6 +175,33 @@ func TestAWSMachinePoolValidateCreate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "Should fail if either roleARN or notifcationARN is set but not both",
+			pool: &AWSMachinePool{
+				Spec: AWSMachinePoolSpec{
+					AWSLifecycleHooks: []AWSLifecycleHook{
+						{
+							RoleARN: aws.String("role-arn"),
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
```

Comment:
I'd add a case for only setting `roleARN`, and another one only setting `notificationARN`

Reply:
done